kong/db/migrations/core/016_280_to_300.lua

```diff
@@ -165,6 +169,14 @@ local function c_migrate_regex_path(coordinator)
         end
       end
 
+      local r = router.new(CACHED_SCHEMA)
+      local exp = _get_expression(route)
+
+      local res, err = r:add_matcher(0, route.id, exp)
+      if not res then
+        log.error("Regex compatibility, route id: %s, err: %s", route.id, err)
```

Comment:
We need to have a bit better error message. Otherwise user will simply be confused about what this is and issues will start flying in.

Could we write a section on "how to fix regex compatibility issues" in `UPGRADE.md` and link the user there in this case?

Reply:
Perhaps we can do the documentation work in another PR, and let this PR focus on validation.